benchmark/kvbench/runtime/etcd_rt.py

```
@@ -76,16 +84,37 @@ def __init__(
             logger.info("Wiping ETCD prefix %s", self.prefix)
             self.client.delete_prefix(self.prefix)
+            self.client.put(init_key, b"1")
+            logger.info("Rank 0 init complete, signaled ready")
+        else:
+            logger.info("Rank %d waiting for rank 0 init...", self.rank)
+            start_time = time.time()
+            while self.client.get(init_key)[0] is None:
+                if time.time() - start_time > init_timeout_sec:
+                    raise TimeoutError(
+                        f"[Rank {self.rank}] Timeout waiting for rank 0 init after {init_timeout_sec}s"
+                    )
```

Ruff (0.15.15)

[warning] Avoid specifying long messages outside the exception class

(TRY003)